src/Tasks/Microsoft.NET.Build.Tasks/EmbedAppNameInHost.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+using System.Text;
+using System.Linq;
+using System.Globalization;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    /// <summary>
+    /// Embeds the App Name into the AppHost.exe  
+    /// </summary>
+    public class EmbedAppNameInHost : TaskBase
+    {
+        private string _modifiedAppHostPath;
+
+        [Required]
+        public string AppHostSourcePath { get; set; }
+
+        [Required]
+        public string AppHostDestinationDirectoryPath { get; set; }
+
+        [Required]
+        public string AppBinaryName { get; set; }
+
+        [Output]
+        public string ModifiedAppHostPath
+        {
+            get { return _modifiedAppHostPath; }
+        }
+
+        protected override void ExecuteCore()
+        {
+            
+            var hostExtension = Path.GetExtension(AppHostSourcePath);
+           
+            var appbaseName = Path.GetFileNameWithoutExtension(AppBinaryName);
+            var bytesToWrite = Encoding.UTF8.GetBytes(AppBinaryName);
+            var destinationDirectory = Path.GetFullPath(AppHostDestinationDirectoryPath);
+            _modifiedAppHostPath = Path.Combine(destinationDirectory, $"{appbaseName}{hostExtension}");
+
+            if ( File.Exists(_modifiedAppHostPath))
```

Comment:
Is it guaranteed that if the apphost.exe exists in the app bin dir, it has been modified? When would this happen? Incremental build?